plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse.go

```diff
@@ -0,0 +1,355 @@
+	// seconds and nanoseconds since January 1, 1970 UTC.
+	// Between Jan 1, 1601 and Jan 1, 1970 there are 11644473600 seconds.
+	return time.Unix(0, (t-116444736000000000)*100)
+}
+
+// createSubscription uses the official 'ctrlX Data Layer API' to create the sse subscription.
+func (s *CtrlXDataLayerSSE) createSubscription(sub *Subscription) (string, error) {
+	sseURL := s.url + subscriptionPath
+
+	payload, id, err := sub.createRequestBody()
+	if err != nil {
+		return "", fmt.Errorf("failed to create subscription %d payload, error: %w", sub.index, err)
+	}
+
+	requestBody := bytes.NewBuffer(payload)
+	req, err := http.NewRequest("POST", sseURL, requestBody)
+	if err != nil {
+		return "", fmt.Errorf("failed to create subscription %d request, error: %w", sub.index, err)
+	}
+
+	req.Header.Add("Authorization", s.tokenManager.Token.String())
+
+	resp, err := s.connection.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to do request to create sse subscription %d, error: %w", sub.index, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 && resp.StatusCode != 201 {
+		return "", fmt.Errorf("failed to create sse subscription %d, status: %q", sub.index, resp.Status)
+	}
+
+	return sseURL + "/" + id, nil
+}
+
+// createSubscriptionAndSseClient creates a sse subscription on the server and
+// initializes a sse client to receive sse events from the server.
+func (s *CtrlXDataLayerSSE) createSubscriptionAndSseClient(sub *Subscription) (*sseclient.SseClient, error) {
+	t, err := s.tokenManager.RequestAuthToken()
+	if err != nil {
+		return nil, err
+	}
+
+	subURL, err := s.createSubscription(sub)
+	if err != nil {
+		return nil, err
+	}
+
+	client := sseclient.NewSseClient(subURL, t.String(), s.InsecureSkipVerify)
+
+	return client, nil
+}
+
+// addMetric writes sse metric into accumulator.
+func (s *CtrlXDataLayerSSE) addMetric(se *sseclient.SseEvent, sub *Subscription) {
+	switch se.Event {
+	case "update":
+		// Received an updated value, that we translate into a metric
+		var d sseEventData
+
+		if err := json.Unmarshal([]byte(se.Data), &d); err != nil {
+			s.acc.AddError(fmt.Errorf("received malformed data from 'update' event"))
+			return
+		}
+		m, err := s.createMetric(&d, sub)
+		if err != nil {
+			s.acc.AddError(fmt.Errorf("failed to create metrics: %v", m))
+			return
+		}
+		s.acc.AddMetric(m)
+	case "error":
+		// Received an error event, that we report to the accumulator
+		var e sseEventError
+		if err := json.Unmarshal([]byte(se.Data), &e); err != nil {
+			s.acc.AddError(fmt.Errorf("received malformed data from 'error' event: %w", err))
+			return
+		}
+		s.acc.AddError(fmt.Errorf("received 'error' event for node: %q", e.Instance))
+	case "keepalive":
+		// Keepalive events are ignored for the moment
+		s.Log.Debug("Received keepalive event")
+	default:
+		// Received a yet unsupported event type
+		s.Log.Debug("Received unsupported event: %q", se.Event)
```

Comment:
Can we please make this an **error** (or at least warning) message? This is something the user should be aware of!